### ISSUE TITLE
vue/script-indent is misbehaving

### ISSUE THREAD

**Issue by MicroDroid** [vuejs/eslint-plugin-vue#344]
> ***I followed the issue template, but, see the last section for a quick look.***

**Tell us about your environment**

Linting a front-end with difference `.vue` and `.js` files with help of SublimeLinter plugin for Sublime Text 3.

* **ESLint Version:** 4.7.2
* **eslint-plugin-vue Version:** 4.2.0
* **Node Version:** 9.3.0

**Please show your full configuration:**

```JavaScript
{
    ...
    "rules": {
        "linebreak-style": [
            "error",
            "unix"
        ],
        "quotes": [
            "error",
            "single"
        ],
        "semi": [
            "error",
            "always"
        ],
        "vue/script-indent": [
            "error",
            "tab",
            {"baseIndent": 1}
        ],
    }
    ...
}
```

**What did you do? Please include the actual source code causing the issue.**
 
Umm, I wrote code? I mean it happened when I tried to use `vue/script-indent`. 

```vue
<script type="text/javascript">
	export default {
		...

		destroyed: function() {
			...
		},

		methods: {
			moment: moment.utc,
			shorten, summarize,
		}, // This is line 101
		...
	};
</script>
```

**What did you expect to happen?**

I expected no lint issues to be reported.

**What actually happened? Please include the actual, raw output from ESLint.**

![image](https://user-images.githubusercontent.com/8523135/34848204-a0a871d4-f726-11e7-9aa5-abacc932c792.png)

From terminal:

```
  101:1    error  Expected indentation of 4 tabs but found 2 tabs               vue/script-indent
```

-------------

Also it's not this only specific case, it happens everywhere in my files. I was able to capture this pattern:

![image](https://user-images.githubusercontent.com/8523135/34848405-4dc801f4-f727-11e7-8273-0d3fe9f251b0.png)

`if`s like this make the line unrelated to the `if` statement required to be indented to `if`'s block level too.

**Comment by mysticatea** [comment]
Thank you for the report.
I confirmed that I can reproduce it: [Oneline demo](https://mysticatea.github.io/vue-eslint-demo/#eJxdUcFqwzAM/RXjU1O8hu3olUEZDHbubvMOaayshsQKsjzShfz7bNKQsJP03pMe4mmUNVqQWh5DTa7nF0PGGy73EFrnWfxEKGflwXkLnrX4BCIkJbi6KDFeqgDvd+Vx+hL7cnaAoUdiYaGpYstinFnDFgIT3sBq0URfs0O/K1bZsGvEbihWbJivLhwsnrGD1PrvXfG8yg3iAia1dHkQbdBb3y6t5yPneohcq1UM13Qs+C0Tu64i9wvqv/2Uy7Hc5CWVpNhCkHqclJxzOqdVqZ8W+HHrc8opszQM1jHSncr5n7x9cwPY1/wLJfuKAlDSIPQEIKc/TKWbmA==)
